### PR TITLE
add juliacall recipe

### PR DESCRIPTION
**Pull request by cjdoris**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by cjdoris** [comment]
Issues with dependencies have been resolved now. Should be good to go.

File: recipes/juliacall/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  name: juliacall
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/j/juliacall/juliacall-{{ version }}.tar.gz
+  sha256: ff81c152831775bbbb26b7752a7825d1de2fd6c154f71ffc22338996b4faac07
+
+build:
+  number: 0
+  skip: True  # [py<35]
```
Comment (beenje) [inline recipes/juliacall/meta.yaml:13]: This is a pure Python package, it should be `noarch: python`. Remove the skip and set a minimum version on Python in host and run `>=3.6` instead.

File: recipes/juliacall/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+
+test:
+  requires:
```
Comment (beenje) [inline recipes/juliacall/meta.yaml:24]: can't you import `juliacall` for test?

File: recipes/juliacall/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  number: 0
+  skip: True  # [py<35]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
```
Comment (beenje) [inline recipes/juliacall/meta.yaml:21]: I guess this package requires Julia to be installed. Wouldn't it make sense to list `julia` in the run requirements?

Comment (cjdoris) [reply recipes/juliacall/meta.yaml]: Currently the package manages its own installation of Julia. Using a conda installation is future work.

File: recipes/juliacall/meta.yaml (hunk 2 above)
Comment (beenje) [reply recipes/juliacall/meta.yaml:24]: Note that conda-build includes an `imports` keyword: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-imports

**Comment by cjdoris** [comment]
Any chance you could run the failed Windows build for me? I'm hoping it's an ephemeral issue, the Julia registry occasionally hiccups. If not I'll dig into it locally.

**Pull request closed without merging**